### PR TITLE
Merge branch 'repo_merge/michael' into 'develop'

### PR DESCRIPTION
**Pull request by mbaddeley**
Merges http://github.com/tiiuae/sc-mesh-secure-deployment into develop. 

* mesh, gateway, and default route set up now done with passing instances through the systemd services:
  - mesh@.service
  - gw@.service
  - default@.service
  - common/scripts/mesh-gw.sh is copied to /usr/local/bin/. instead of run-gw.sh in /usr/bin
  - common/scripts/mesh-default.sh is copied into /usr/local/bin.
* mesh-ibss.sh is used instead of mesh_init.sh
  - args ${10} and ${11} now passed for wlan interface and phy
  - IFF wlan interface specified then wpa_supplicant DOESN'T run in background (-B) as it's run as a service
  - IFF wlan interface specified then wpa_supplicant is not killed as it kills the gateway on the on-board wlan.

Has been tested with 1 gw and 2 regular mesh nodes:

* batctl n --> shows other nodes at L2
* ping 10.20.15.x --> successful pings between all nodes
* ping 8.8.8.8 --> successful ping from regular mesh nodes through gw
* ping www.google.com --> successful ping from regular mesh nodes through gw
* ssh 10.20.15.x --> successful ssh to all nodes (within mesh)

### CODE REVIEW

File: common/scripts/mesh-ibss.sh [hunk 1]
```diff
@@ -3,31 +3,43 @@
...
+# <mode> <ip> <mask> <AP MAC> <key> <essid> <freq> <txpower> <country> <interface>
+
 echo "Solving wifi device name.."
-wifidev=$(iw dev | awk '$1=="Interface"{print $2}')
+if [[ -z "${10}" ]]; then
+  wifidev=$(iw dev | awk '$1=="Interface"{print $2}')
+else
+  wifidev=${10}
+  phyname=${11}
+fi
```
Comment (joenpera) [inline common/scripts/mesh-ibss.sh:35]: phyname not set in if-branch

File: common/scripts/mesh-ibss.sh [hunk 2]
```diff
@@ -3,31 +3,43 @@
...
     echo "	<essid>"
     echo "	<freq>"
+    echo "	<txpower>"
+    echo "	<country>"
+    echo "	<interface>"
+    echo "	<phyname>"
     echo
     echo "example:"
-    echo "sudo mesh.sh mesh 192.168.1.2 255.255.255.0 00:11:22:33:44:55 1234567890 mymesh2 5220 30 fi"
+    echo "sudo mesh.sh mesh 192.168.1.2 255.255.255.0 00:11:22:33:44:55 1234567890 mymesh2 5220 wlan1 phy1"
```
Comment (joenpera) [inline common/scripts/mesh-ibss.sh:21]: example overwritten "30 fi"

File: common/scripts/mesh-ibss.sh [hunk 3]
```diff
@@ -65,7 +83,7 @@ EOF
 
       echo "$wifidev down.."
       iw dev $wifidev del
-      iw phy phy0 interface add $wifidev type ibss
+      iw phy $phyname interface add $wifidev type ibss
```
Comment (joenpera) [inline common/scripts/mesh-ibss.sh:86]: $phyname not set in all cases

File: common/scripts/mesh-default.sh [hunk 4]
```diff
@@ -0,0 +1,3 @@
+#!/bin/bash
```
Comment (joenpera) [inline common/scripts/mesh-default.sh:1]: This is setting default gateway.  Should it be renamed to mesh-default-gw.sh

Comment (mbaddeley) [reply common/scripts/mesh-default.sh]: Agreed

**Review by joenpera — APPROVED** [review]
common/scripts/mesh-ibss.sh is ok

**Comment by mbaddeley** [comment]
@pemanty any comments? Or am I ok to merge to develop?

**Review by pemanty — APPROVED** [review]
common/scripts/mesh-ibss.sh is ok
Good to go.

**Pull request merged**